src/Nethermind/Nethermind.Optimism/OptimismCostHelper.cs

```diff
@@ -290,4 +324,31 @@ uint setNextHash(uint ip, ref Span<uint> ht)
     }
 
     internal static UInt256 ComputeGasUsedFjord(UInt256 estimatedSize) => estimatedSize * GasCostOf.TxDataNonZeroEip2028 / BasicDivisor;
+
+    // https://specs.optimism.io/protocol/jovian/exec-engine.html#scalar-loading
+    private UInt256 GetDaFootprintScalar(IWorldState? worldState)
+    {
+        ReadOnlySpan<byte> span = worldState is null ? ReadOnlySpan<byte>.Empty : worldState.Get(_daFootprintGasScalarSlot);
+
+        const int scalarPosition = 12;
+        if (span.Length < scalarPosition + sizeof(UInt16))
+            return DaFootprintScalarDefault;
+
+        var scalar = ReadUInt16BigEndian(span[scalarPosition..]);
+        return scalar == 0 ? DaFootprintScalarDefault : scalar;
+    }
+
+    // https://specs.optimism.io/protocol/jovian/l1-attributes.html
+    private static UInt256 GetDaFootprintScalar(Block block)
+    {
+        var firstTx = block.Transactions.FirstOrDefault();
+        if (firstTx?.Type is not TxType.DepositTx)
+            return DaFootprintScalarDefault;
+
+        if (firstTx.Data.Length < 178)
+            return DaFootprintScalarDefault;
+
+        var scalar = ReadUInt16BigEndian(firstTx.Data.Span[176..178]);
+        return scalar == 0 ? DaFootprintScalarDefault : scalar;
+    }
```

Comment:
Added 2 options to calculate `daFootprintGasScalar` as [per documentation](https://specs.optimism.io/protocol/jovian/exec-engine.html#scalar-loading).

Not sure which one is better, but version taking if from block transactions directly doesn't require state, so has more diverse usage options. Can also try to fetch from both sources before falling back to default value.

Reply:
Also, what does op-geth do in this scenario? Maybe the better approach is just to copy what they're doing.

Reply:
They also extract it from block tx data:
https://github.com/ethereum-optimism/op-geth/blob/aa940538653c59df97b432bc24a2a4e4b17a06a8/core/types/rollup_cost.go#L547C6-L547C33